src/new-client/src/App/Footer/StatusButton/serviceInstanceCollection.ts

```diff
@@ -0,0 +1,72 @@
+import { ServiceStatus, ServiceConnectionStatus } from "services/connection"
+
+export interface ServiceInstanceStatus {
+  serviceType: string
+  serviceId: string
+  timestamp: number
+  serviceLoad: number
+  isConnected: boolean
+}
+
+export class ServiceInstanceCollection {
+  private readonly serviceMap: Map<string, ServiceInstanceStatus> = new Map()
+
+  constructor(public readonly serviceType: string) {}
+
+  update(update: ServiceInstanceStatus) {
+    this.serviceMap.set(update.serviceId, update)
+    return this
+  }
+
+  private getServiceInstances() {
+    return Array.from(this.serviceMap.values())
+  }
+
+  getServiceNumberOfInstances() {
+    return this.getServiceInstances().filter((x) => x.isConnected).length
+  }
+}
+
+export interface IServiceStatusCollection {
+  getServiceNumberOfInstances: (serviceType: string) => number | undefined
+}
+export class ServiceCollectionMap implements IServiceStatusCollection {
+  private readonly serviceInstanceCollections = new Map<
+    string,
+    ServiceInstanceCollection
+  >()
+
+  add(service: string, serviceInstanceCollection: ServiceInstanceCollection) {
+    this.serviceInstanceCollections.set(service, serviceInstanceCollection)
+    return this
+  }
+
+  getServiceNumberOfInstances(serviceType: string) {
+    const x = this.serviceInstanceCollections.get(serviceType)
+
+    if (x) {
+      return x.getServiceNumberOfInstances()
+    }
+
+    return undefined
+  }
+
+  getStatusOfServices(): ServiceConnectionInfo {
```

Comment:
As it happens, I've spent a good chunk of time today looking at this code in the original client.   Think you can make this method a little simpler with fromEntries and our mapObject utility: 

```
  return mapObject(
     Object.fromEntries(this.serviceInstanceCollections),
    (instanceCollection) =>  ({
      serviceType: instanceCollection.serviceType,
      connectedInstanceCount: instanceCollection.getServiceNumberOfInstances(),
      connectionStatus: instanceCollection.getServiceNumberOfInstances()
        ? ServiceConnectionStatus.CONNECTED
        : ServiceConnectionStatus.DISCONNECTED,
   })
  )
```

The keys of the `this.serviceInstanceCollections` Map are the serviceTypes:

https://github.com/AdaptiveConsulting/ReactiveTraderCloud/pull/1742/files#diff-431d7047cc2d8c8daac07b0dca1679abc5e0d376488d73c194d6addc8299b500R104
)